libpod/runtime_img.go

```diff
@@ -96,30 +173,35 @@ func (r *Runtime) PullImage(imgName string, allTags bool, signaturePolicyPath st
 
 	// PullImage copies the image from the source to the destination
 	var (
-		images []string
+		images       []string
+		fromRegistry bool
 	)
 
-	if signaturePolicyPath == "" {
-		signaturePolicyPath = r.config.SignaturePolicyPath
+	signaturePolicyPath := r.config.SignaturePolicyPath
+	if options.SignaturePolicyPath != "" {
+		signaturePolicyPath = options.SignaturePolicyPath
 	}
 
-	sc := common.GetSystemContext(signaturePolicyPath, "")
+	sc := common.GetSystemContext(signaturePolicyPath, options.AuthFile)
 
 	srcRef, err := alltransports.ParseImageName(imgName)
 	if err != nil {
-		defaultName := DefaultRegistry + imgName
-		srcRef2, err2 := alltransports.ParseImageName(defaultName)
-		if err2 != nil {
-			return errors.Errorf("error parsing image name %q: %v", defaultName, err2)
+		// could be trying to pull from registry with short name
+		images, err = getRegistriesToTry(imgName)
```

Comment:
This function successfully calls `alltransports.ParseImageName` on each of the strings returned in `images`, and then discards that, and the same parsing happens later when setting `pullRef`. It would be more efficient, and cleaner, to return a `[]struct{storageReference string, pullRef types.ImageReference}`.  Of perhaps even `[]struct{destRef, srcRef types.ImageRerence}`.

Reply:
fixed. Using a struct that holds srcRef and dstRef now.